plugins/obs-outputs/rtmp-stream.c

```diff
@@ -1234,6 +1424,24 @@ static void *connect_thread(void *data)
 		stream->start_dts_offset = get_ms_time(&packet, packet.dts);
 	}
 
+	// HDR streaming disabled for AV1 and HEVC
+	obs_output_t *context = stream->output;
+	obs_encoder_t *vencoder = obs_output_get_video_encoder(context);
+	const char *codec = obs_encoder_get_codec(vencoder);
+	if (strcmp(codec, "hevc") == 0 || strcmp(codec, "av1") == 0) {
+		video_t *video = obs_get_video();
+		const struct video_output_info *info =
+			video_output_get_info(video);
+		enum video_format format = info->format;
+		enum video_colorspace colorspace = info->colorspace;
+
+		if (format >= VIDEO_FORMAT_I010) { // 10-bits video and more
+			obs_output_signal_stop(stream->output,
+					       OBS_OUTPUT_HDR_DISABLED);
+			return NULL;
+		}
+	}
```

Comment:
I don't understand, it does not stop on HDR colorspace but on 10-bit format.

Reply:
Yeah this would need clarification on whether or not only HDR colourspaces should be disallowed, or 10-bit video in its entirety (since 10-bit SDR is possible).

Reply:
Also don’t treat an enum like an integer - the order is not guaranteed and we could choose to change or reorder the entries - explicitly check for specific entries or use a `switch` statement.

Reply:
Yes good call, we will change this to check on colorspace instead.  We have a bug with 10-bit AV1 streams but I will add that to Known Issues instead of guarding the behaviour here.

Reply:
Done